notebooks/working/Causal_Graph_Tim_Data.py

```diff
@@ -0,0 +1,446 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.3.4
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# +
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from causalgraphicalmodels import CausalGraphicalModel, StructuralCausalModel
+import pylogit
+from collections import OrderedDict
+import pylogit as cm
+from functools import reduce
+
+
+
+import os
+os.listdir('.')
+# -
+
+data = pd.read_csv('spring_2016_all_bay_area_long_format_plus_cross_bay_col.csv')
+data.columns
+
+data['mode_id'].unique()
+
+# +
+X_columns = ['total_travel_time',
+       'total_travel_cost', 'total_travel_distance', 
+             'cross_bay', 'household_size', 'num_kids', 
+              'cars_per_licensed_drivers', 
+             'gender'
+             
+            ]
+
+y_column = data['mode_id']
+
+# +
+sprinkler = CausalGraphicalModel(
+    nodes=["Travel_Time", "Travel_Distance", "Travel_Cost", "Cross_Bay_Bridge", "HH_Size", "num_of_kids_household",
+          "Autos_per_licensed_drivers", "Gender", "Mode_Choice"],
+    edges=[
+        ("Travel_Time", "Mode_Choice"), 
+        ("Travel_Distance", "Mode_Choice"), 
+        ("Travel_Cost", "Mode_Choice"), 
+        ("Cross_Bay_Bridge", "Mode_Choice"), 
+        ("HH_Size", "Mode_Choice"), 
+        ("num_of_kids_household", "Mode_Choice"), 
+        ("Autos_per_licensed_drivers", "Mode_Choice"), 
+        ("Gender", "Mode_Choice"), 
+        
+        
+        ("Travel_Distance", "Travel_Time"), 
+        ("Travel_Distance", "Travel_Cost"), 
+        ("Travel_Distance", "Cross_Bay_Bridge"), 
+        ("HH_Size", "Travel_Distance"), 
+
+        
+        
+#         ("Travel_Time", "Mode_Choice"), 
+#         ("Travel_Time", "Mode_Choice"), 
+
+        
+        
+    ]
+)
+
+# draw return a graphviz `dot` object, which jupyter can render
+sprinkler.draw()
+# -
+
+# # MNL specification
```

Comment:
Currently, you go directly from the creation of a causal graph (i.e. step 2 in section 5.2 of the causal inference paper I have with Joan), to MNL estimation (i.e step 5).

So far step 3 (attempted falsification of the causal graph) and step 4 (identification of the causal effect) are missing.

Please include code here for those two steps.